### PR TITLE
New package: armcord-3.1.4

### PR DESCRIPTION
**Pull request by Zoh-j02r**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**
 

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686*

🤮electron

### CODE REVIEW

**Comment by Duncaen** [comment]
How is that supposed to work, you are not building electron from source or use the system wide electron, where are the armv7l-musl binaries coming from?

**Comment by paper42** [comment]
> You are breaking Discord ToS by using ArmCord, but no one has been banned from using it or any of the client mods included.

And this is just a wrapper around the discord web app, so I don't think packaging this is a good idea.

**Comment by Zoh-j02r** [comment]
> How is that supposed to work, you are not building electron from source or use the system wide electron, where are the armv7l-musl binaries coming from?

Sorry, I didnt realize Electron-builder was pulling a Electron zip, I set it to use the system Electron now, but it is not set for other archtectures but x64. I dont know how to set it to use other archtectures system electron builds or if it is not even possible.

**Comment by Zoh-j02r** [comment]
> > You are breaking Discord ToS by using ArmCord, but no one has been banned from using it or any of the client mods included.
> 
> And this is just a wrapper around the discord web app, so I don't think packaging this is a good idea.

It is, but it uses a newer version of Electron and  can be built for other architectures and is working in features the Discord team it not even willing to work on  ArmCord/ArmCord#305 and [😃](https://github.com/ArmCord/ArmCord/issues/161#issuecomment-1189970980).

>

File: srcpkgs/armcord/files/armcord.desktop [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=ArmCord
+Comment=Custom client designed to enhance Discord experience"
```
Comment (Vendicated) [inline srcpkgs/armcord/files/armcord.desktop:3]:
```suggestion
Comment=Custom client designed to enhance your Discord experience
```

File: srcpkgs/armcord/template [hunk 2]
```diff
@@ -0,0 +1,55 @@
+
+# Template file for 'armcord'
+pkgname=armcord
+version=3.1.4
+revision=1
+archs="x86_64* i686* aarch64* armv7l*"
+hostmakedepends="nodejs electron19 git"
```
Comment (Vendicated) [inline srcpkgs/armcord/template:7]: optimally, electron21+ should be used if available, for best support

**Comment by Zoh-j02r** [comment]
OMG what have I done

**Comment by yurifuko** [comment]
I'd actually really like this, a more updated and convenient way of using discord in a separate app that supports theming, has tracker blocking, updated electron and more features being worked on etc.

I don't think it technically breaking the ToS is really significant because there are `6cord` and `cordless` packages (weirdly enough since they're deprecated and actually have a risk that comes with using them it seems)

so why not? impatiently waiting for activity on this PR lol

**Comment by paper42** [comment]
> I don't think it technically breaking the ToS is really significant because there are `6cord` and `cordless` packages (weirdly enough since they're deprecated and actually have a risk that comes with using them it seems)

And I think we should remove them.

> so why not? impatiently waiting for activity on this PR lol

Because electron is hard to maintain, only available for x86_64* and the more applications that use it, the more issues there are going to be with updates. I think this is more appropriate to be packaged in a container like flatpak which won't have these issues.

File: srcpkgs/armcord/template (hunk 2 above)
Comment (paper42) [reply srcpkgs/armcord/template:6]: We don't have electron21 unfortunately

**Comment by Zoh-j02r** [comment]
> Because electron is hard to maintain, only available for x86_64* and the more applications that use it, the more issues there are going to be with updates. I think this is more appropriate to be packaged in a container like flatpak which won't have these issues.

This template is very similar to [Rocket.Chat-Desktop](https://github.com/void-linux/void-packages/pull/24664), which has been merged and is being updated with no problems thanks to electron-builder. Even when forcing armcord to use electron19 to build, there was not errors or crashes.

**Comment by paper42** [comment]
There is now an armcord flatpak: https://tech.lgbt/@orowith2os/110053383155464246

**Comment by Zoh-j02r** [comment]
Oh wow, that is great. I now understand why using flatpak is more appropriate for this case

**Comment by paper42** [comment]
let's close this, ArmCord is now available through flatpak and it would probably only cause us an headache maintaining it in void repositories

**Pull request closed without merging**